File: frontend/src/api/uso.ts
import { onUnauthorized } from "@/lib/session"
export interface UsoAgg {
  chamadas: number
  input_tokens: number
  output_tokens: number
  tokens: number
  custo_usd: number
}

export interface UsoResumo {
  total: UsoAgg
  mes: UsoAgg
}

export interface ClaudeModelo {
  model: string
  input: number
  output: number
  cache: number
}

export interface ClaudeCodeUso {
  disponivel: boolean
  atualizado_em: string | null
  sessoes: number
  mensagens: number
  tokens_input: number
  tokens_output: number
  tokens_total: number
  cache_tokens: number
  por_modelo: ClaudeModelo[]
  hoje: { date: string | null; tokens: number } | null
}

async function request<T>(path: string): Promise<T> {
  const res = await fetch(`/api/v1${path}`, { credentials: "include" })
  if (!res.ok) {
    if (res.status === 401) onUnauthorized()
    const data = await res.json().catch(() => ({}))
    throw new Error(data?.detail ?? `Erro ${res.status}`)
  }
  return res.json()
}

export const usoApi = {
  resumo: () => request<UsoResumo>("/uso"),
  claudeCode: () => request<ClaudeCodeUso>("/uso/claude-code"),
}

File: backend/app/schemas/uso.py
from pydantic import BaseModel


class UsoAgg(BaseModel):
    chamadas: int
    input_tokens: int
    output_tokens: int
    tokens: int
    custo_usd: float


class UsoResumo(BaseModel):
    total: UsoAgg
    mes: UsoAgg


class ClaudeModelo(BaseModel):
    model: str
    input: int
    output: int
    cache: int


class ClaudeHoje(BaseModel):
    date: str | None = None
    tokens: int = 0


class ClaudeCodeUso(BaseModel):
    disponivel: bool
    atualizado_em: str | None = None
    sessoes: int
    mensagens: int
    tokens_input: int
    tokens_output: int
    tokens_total: int
    cache_tokens: int
    por_modelo: list[ClaudeModelo]
    hoje: ClaudeHoje | None = None
